packages/cli/src/cliEntry.js

```diff
@@ -162,6 +164,27 @@ async function setupAndRun() {
 
   setProjectDir(ctx.root);
 
+  try {
+    // New version check must occur before `commander.parse` to ensure that
+    // the message of a new release happens before anything else.
+    const {version: currentVersion} = require(path.join(
+      ctx.root,
+      'node_modules/react-native/package.json',
+    ));
```

Comment:
Please use `require.resolve` or `resolveNodeModuleDir` helper for that as in future it may happen that there's no `node_modules`, something like this should do:
```js
const {version: currentVersion} = require(
  path.join(
    resolveNodeModuleDir(ctx.root, 'react-native'), 
    'package.json'
  )
);
```